packages/fxa-auth-server/lib/customs.js

```diff
@@ -4,192 +4,150 @@
-        return { block: false };
-      };
-      const noop = async function () {};
-      this.api = {
-        check: noblock,
-        checkAuthenticated: noblock,
-        checkIpOnly: noblock,
-        failedLoginAttempt: noop,
-        passwordReset: noop,
-        close: noop,
-      };
-    } else {
-      this.api = new CustomsAPI(url, { timeout: 3000 });
-    }
+  async checkAuthenticated(request, uid, action) {
+    const result = await this.makeRequest('/checkAuthenticated', {
+      ...this.sanitizePayload({
+        action,
+        ip: request.app.clientAddress,
+        uid,
+      }),
+    });
+
+    this.optionallyReportStatsD('request.checkAuthenticated', action, result);
+    return this.handleCustomsResult(request, result);
   }
 
-  Customs.prototype.check = async function (request, email, action) {
-    const result = await this.api.check({
-      ip: request.app.clientAddress,
-      email: email,
-      action: action,
-      headers: request.headers,
-      query: request.query,
-      payload: sanitizePayload(request.payload),
+  async checkIpOnly(request, action) {
+    const result = await this.makeRequest('/checkIpOnly', {
+      ...this.sanitizePayload({
+        action,
+        ip: request.app.clientAddress,
+      }),
     });
 
-    optionallyReportStatsD('request.check', action, result);
+    this.optionallyReportStatsD('request.checkIpOnly', action, result);
+    return this.handleCustomsResult(request, result);
+  }
 
-    return handleCustomsResult(request, result);
-  };
+  async flag(ip, info) {
+    await this.makeRequest('/failedLoginAttempt', {
+      ...this.sanitizePayload({
+        ip,
+        email: info.email,
+        errno: info.errno || this.error.ERRNO.UNEXPECTED_ERROR,
+      }),
+    });
+  }
 
-  function optionallyReportStatsD(name, action, options = {}) {
-    if (statsd) {
-      const { block, unblock, suspect, blockReason } = options;
+  async reset(email) {
+    await this.makeRequest('/passwordReset', {
+      ...this.sanitizePayload({ email }),
+    });
+  }
 
-      // The action and block values will always be reported, all others are optional
-      const tags = { action, block };
+  /**
+   * Remove sensitive fields from the payload before sending to customs.
+   *
+   * @param payload
+   * @return {*}
+   */
+  sanitizePayload(payload) {
+    if (!payload) {
+      return;
+    }
 
-      if (options.unblock !== undefined) {
-        tags['unblock'] = unblock;
-      }
+    const clonePayload = { ...payload };
+    const fieldsToOmit = ['authPW', 'oldAuthPW', 'paymentToken'];
+    fieldsToOmit.forEach((name) => delete clonePayload[name]);
 
-      if (options.suspect !== undefined) {
-        tags['suspect'] = suspect;
-      }
+    return clonePayload;
+  }
 
-      if (options.blockReason !== undefined) {
-        tags['blockReason'] = blockReason;
-      }
+  optionallyReportStatsD(name, action, options = {}) {
+    if (!options) {
+      return;
+    }
```

Comment:
I can't tell if this is intentional or a small issue, but with `options` having a default value of `{}` the predicate will be false when `options` has the default value.